seismic/inventory/add_time_corrections.py

```diff
@@ -0,0 +1,154 @@
+#! /usr/bin/env python
+"""
+Add GPS clock time correction csv_data into inventory file to get a modified station xml file
+
+CreationDate:
+    24/02/2020
+
+Developer:
+    fei.zhang@ga.gov.au
+
+"""
+
+import os
+import sys
+from obspy import read_inventory
+from obspy import Inventory
+from obspy.core.inventory import Network
+from obspy.core.util import AttribDict
+
+
+def get_csv_correction_data(path_csvfile):
+    """
+    Read in the csv data from an input file, get the network_code, station_code, csv_data
+
+    Args:
+        path_csvfile: input csv file in /g/data/ha3/Passive/SHARED_DATA/GPS_Clock/corrections/
+        $ head 7D.DE43_clock_correction.csv
+        net,sta,date,clock_correction
+        7D,DE43,2012-11-27,1.0398489013215846
+        7D,DE43,2012-11-28,0.9408504322549281
+        7D,DE43,2012-11-29,0.8418519631882714
+        7D,DE43,2012-11-30,0.7428534941216148
+        7D,DE43,2012-12-01,0.6438550250549583
+
+    Returns: (network_code, station_code, csv_data)
+
+    """
+    with open(path_csvfile, "r") as csvfid:
+        all_lines = csvfid.readlines()
+
+    print("The csv file length", len(all_lines))
+    print("the first line", all_lines[0])
+    print("the last  line", all_lines[-1])
+
+    line2 = all_lines[1]
+    print("line2=", line2)
+
+    my_items = line2.split(",")
+
+    network_code = my_items[0]  # network_code = "7D"
+    station_code = my_items[1]  # station_code = "DE43"
+
+    return (network_code, station_code, all_lines)
+
+
+def add_gpscorrection_into_stationxml(csv_file, input_xml, out_xml=None):
+    """
+    read in the correction CSV data from a file, get the station metadata node from input_xml file, 
+    then add the CSV data into the station xml node to write into out_xml
+    Args:
+        csv_file: input csv file with correction data
+        input_xml: input original stationXML file which contains the metadata for the network and station of csv_file
+        out_xml:  Directory of the output xml file
+
+    Returns:
+        full path of the output xml file
+
+    """
+
+    ns = "https://github.com/GeoscienceAustralia/hiperseis/xmlns/1.0"
+
+    (net, sta, csv_data) = get_csv_correction_data(csv_file)
+
+    # path2_myxml = "/home/feizhang/Githubz/hiperseis/tests/testdata/7D_2012_2013.xml"
+    my_inv = read_inventory(input_xml)
```

Comment:
It might be a good idea to explicitly pass the format, i.e. `format='stationxml'`, just in case someone passes a different format to this script. This way the function can raise an error if the input file format is not stationxml.

Reply:
I don't see any change here, what did you decide to do?

Reply:
updated the code to:
 read_inventory(path2xml, format='STATIONXML')